ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/BlockBroadcaster.java

```diff
@@ -45,7 +47,13 @@ public void unsubscribePropagateNewBlocks(final long id) {
 
   public void propagate(final Block block, final Difficulty totalDifficulty) {
     blockPropagatedSubscribers.forEach(listener -> listener.accept(block, totalDifficulty));
-    final NewBlockMessage newBlockMessage = NewBlockMessage.create(block, totalDifficulty);
+    final NewBlockMessage newBlockMessage;
+    try {
+      newBlockMessage = NewBlockMessage.create(block, totalDifficulty, this.maxMessageSize);
+    } catch (final IllegalArgumentException e) {
+      LOG.error("Failed to create block", e);
+      return;
+    }
```

Comment:
Original `send` in EthPeers catches the error (when message size > limit) instead of throwing it up. Would it better to throw the error? I wasn't too sure on the original intention when they caught the error instead of throwing it.